src/lib/actions/getAnnouncementsUsingSubgraph/getAnnouncementsUsingSubgraph.ts

```diff
@@ -0,0 +1,67 @@
+import { GraphQLClient } from 'graphql-request';
+import type { AnnouncementLog } from '../getAnnouncements/types';
+import {
+  convertSubgraphEntityToAnnouncementLog,
+  fetchPages
+} from './subgraphHelpers';
+import type {
+  GetAnnouncementsUsingSubgraphParams,
+  GetAnnouncementsUsingSubgraphReturnType,
+  SubgraphAnnouncementEntity
+} from './types';
+
+async function getAnnouncementsUsingSubgraph({
+  subgraphUrl,
+  filter = '',
+  pageSize = 1000
```

Comment:
I believe alchemy supports page size of 10k.

Reply:
But maybe it's better to just leave it at 1K, and just make a note of it. Because other subgraph providers usually only support 1k.

Reply:
cool updated with docs:

[b29c692](https://github.com/ScopeLift/stealth-address-sdk/pull/78/commits/b29c69209a272fda215ee7c6ddeb8b3a7fd16807)